libs/sdk-ui-dashboard/src/presentation/dragAndDrop/DragLayerPreview/HeightResizerDragPreview.tsx

```diff
@@ -0,0 +1,100 @@
+// (C) 2021-2022 GoodData Corporation
+import React, { useEffect, useState } from "react";
+
+import { HeightResizer } from "../Resize/HeightResizer";
+import clamp from "lodash/clamp";
+import { DragPreviewProps, ReachedHeightResizingLimit } from "./types";
+import { fluidLayoutDescriptor } from "@gooddata/sdk-ui-ext";
+
+import { HeightResizerDragItem } from "../types";
+import { useResizeHandlers } from "../LayoutResizeContext";
+
+export type IHeightResizerDragPreviewProps = DragPreviewProps<HeightResizerDragItem>;
+
+export const HeightResizerDragPreview = (props: IHeightResizerDragPreviewProps) => {
+    const { item, initialOffset, differenceFromInitialOffset, documentDimensions } = props;
+    const [hasReachedLimit, setReachedLimit] = useState<ReachedHeightResizingLimit>("none");
+    const { toggleHeightLimitReached } = useResizeHandlers();
+    useEffect(() => {
+        toggleHeightLimitReached(hasReachedLimit);
+    }, [hasReachedLimit, toggleHeightLimitReached]);
+
+    const currentOffsetY = differenceFromInitialOffset.y;
+
+    const currentUnlimitedHeightGR = getNewHeightGR(
+        item.widgetHeights,
+        currentOffsetY,
+        documentDimensions.scrollTop,
+        item.initialScrollTop,
+    );
+    const hasNowReachedLimit = hasHeightReachedLimit(currentUnlimitedHeightGR, item.minLimit, item.maxLimit);
+
+    if (hasNowReachedLimit !== hasReachedLimit) {
+        setReachedLimit(hasNowReachedLimit);
```

Comment:
This is suspicious: a side effect in render. Should this be an useEffect instead?